core/lib/zksync_core/src/lib.rs

```diff
@@ -758,15 +758,15 @@ async fn add_state_keeper_to_task_futures(
         .build()
         .await
         .context("failed to build state_keeper_pool")?;
-    let next_priority_id = state_keeper_pool
-        .access_storage()
-        .await
-        .unwrap()
-        .transactions_dal()
-        .next_priority_id()
-        .await;
-    let mempool = MempoolGuard::new(next_priority_id, mempool_config.capacity);
-    mempool.register_metrics();
+    let mempool = {
```

Comment:
I'd say that it's the fairly idiomatic thing in Rust. Alternatively, we can call `drop(connection)`, but personally I like scoped blocks more (since the "lifetime" of the connection is indented).

Reply:
I look at it from the different angle: you don't even need to understand that something intermediate is instantiated and dropped there. The code tells you that _nothing but mempool_ outlives this block, and if you don't know specifics, it's generally all the information you should have, as if this was a separate function.